docs/ref/models/lookups.txt

```diff
@@ -94,6 +94,19 @@ following methods:
     ``compiler.compile(expression)`` should be used. The ``compiler.compile()``
     method will take care of calling vendor-specific methods of the expression.
 
+    ``kwargs`` may be used to supply extra context to the ``as_sql``-method
+    for that particular call. This can be useful for Expressions in
+    particular, where ``template`` can be overridden and variables intended
+    for interpolating the ``template``-string can be passed. 3rd-party
+    backends may have different SQL-syntax so they can pass these differences
+    through via ``kwargs``.
+
+    Example::
+
+        self.as_sql(compiler, connection,
+            template="%(function)s(%(arg)s)", function='Print', arg='Hello World'
+        )
```

Comment:
How's this? I'm struggling to find an example that would add it's own private interpolation variables within the template and then supply them as a kwarg rather than just embedding the data within the template to begin with. For that reason I'm not so sure that it should be highlighted here as a main feature anymore.

```
Example::

# Concat function implementation for sqlite
def as_sqlite(self, compiler, connection):
    return self.as_sql(compiler, connection, template='%(expressions)s', arg_joiner=' || ')
```